strategies/affected/src/main/java/org/arquillian/smart/testing/strategies/affected/ClassFileIndex.java

```diff
@@ -114,66 +112,59 @@ private void replaceVertex(JavaElement newClass) {
 
     private void updateJavaElementWithImportReferences(JavaElement javaElementParentClass,
         String[] imports) {
+
         for (String importz : imports) {
+
+            if (addImport(javaElementParentClass, importz)) {
+                if (filter.shouldBeIncluded(importz)) {
+                    JavaClass javaClass = builder.getClass(importz);
+                    if (javaClass != null) {
+                        updateJavaElementWithImportReferences(javaElementParentClass, javaClass.getImports());
+                    }
+                }
+            }
+        }
+    }
+
+    private boolean addImport(JavaElement javaElementParentClass, String importz) {
+
+        if (!importz.startsWith("java")) {
+
             JavaElement importClass = new JavaElement(importz);
             if (!importClass.equals(javaElementParentClass)) {
                 if (!graph.containsVertex(importClass)) {
                     graph.addVertex(importClass);
                 }
+
+                // This condition is required because we are flattering imports in graph
+                // This operation is fast since it is a containsKey call in a Map instance
+                if (!graph.containsEdge(javaElementParentClass, importClass)) {
+                    graph.addEdge(javaElementParentClass, importClass);
+                    return true;
+                }
                 graph.addEdge(javaElementParentClass, importClass);
             }
         }
-    }
 
-    // I don't remove these methods for now because might be used if we decide to resolve transitive imports
-    /**private JavaElement findOrCreateJavaClass(String classname) {
-        JavaElement clazz = findClass(classname);
-        if (clazz == null) {
-            JavaClass javaClass = builder.getClass(classname);
-            clazz = new JavaElement(javaClass);
-            if (javaClass.locatedInClassFile()) {
-                addToIndex(clazz, javaClass.getImports());
-            }
-        }
-        return clazz;
+        return false;
     }
 
-    private JavaElement findClass(String classname) {
-        for (JavaElement jClass : graph.vertexSet()) {
-            if (jClass.getClassName().equals(classname)) {
-                return jClass;
-            }
-        }
-        return null;
-    }**/
 
     public Set<String> findTestsDependingOn(Set<File> classes) {
-        final Set<JavaElement> javaClasses = classes.stream()
+        return classes.stream()
             .map(javaClass -> JavaToClassLocation.transform(javaClass, globPatterns))
```

Comment:
Probably we should change the variable name since it is not the glob patterns that comes from surefire, this is a glob pattern used to detect if a `File` is a test or not https://github.com/arquillian/smart-testing/blob/master/strategies/affected/src/main/java/org/arquillian/smart/testing/strategies/affected/detector/FileSystemTestClassDetector.java

So if we want to remove it which is possible and correct me if I am wrong I can use the `TestVerifier` class instead of glob resolution.